include/wx/hwinfo.h

```diff
@@ -0,0 +1,40 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/hwinfo.h
+// Purpose:     declaration of the wxSystemHardwareInfo class
+// Author:      Blake Madden
+// Created:     2024-07-09
+// Copyright:   (c) 2024 Blake Madden
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_SYSTEMHWINFO_H_
+#define _WX_SYSTEMHWINFO_H_
+
+#include "wx/string.h"
+#include "wx/utils.h"
+#include "wx/power.h"
+
+// ----------------------------------------------------------------------------
+// wxSystemHardwareInfo
+// ----------------------------------------------------------------------------
+
+// Information about the system's hardware
+class WXDLLIMPEXP_BASE wxSystemHardwareInfo
```

Comment:
I think there are 2 choices that generally make sense:

1. Make this a namespace with functions inside it.
2. Make this a class with static `Get()` function and non-static accessors.

The latter makes sense if we ever want/need to cache some information, e.g. if `GetPhysicalMemory()` is called, we could also get the result of `GetFreePhysicalMemory()` together with it.

I'm not sure if we really need it, but having a class with just static functions doesn't seem to have any advantages compared to the first approach, so in this case I'd just use a namespace.